### DISCUSSION TITLE
configure poetry version range

### DISCUSSION THREAD

**Discussion by anarkiwi** [renovatebot/renovate#18036] [Request Help]
### What would you like Renovate to be able to do?

As poetry is evolving, we would like to be able to have Renovate use (or be limited) to a a version range. 

As others are we're gradually migrating our projects from setuptools style to setuptools-free. This will take a while. In the meantime, poetry 1.1.x and 1.2.x handle having setuptools differently which makes the migration hard to manage (we'd like to use 1.1.x for the moment which works fine at this stage, but 1.2.x is relatively unstable for us - Renovate appears to use 1.2.x).

### If you have any ideas on how this should be implemented, please tell us here.

It would be great to be able to pin Renovate to use just the latest 1.1.x (or 1.2.x, etc). Then "poetry update --lock" works just fine (right now that's what we do manually creating PRs with poetry 1.1.x installed - we'd like Renovate to do the same).

### Is this a feature you are interested in implementing yourself?

Yes

**Comment by viceice** [comment]
this should already be possible, checkout `constraints` in the docs.

**Comment by anarkiwi** [comment]
Thanks! I tried it, and it did not work the way I expected.

installing v2 tool python v3.10.7
Using prebuild python
Requirement already satisfied: pip in
/opt/buildpack/tools/python/3.10.7/lib/python3.10/site-packages (22.2.2)
Files removed: 1
linking tool python v3.10.7
Python 3.10.7
pip 22.2.2 from
/opt/buildpack/tools/python/3.10.7/lib/python3.10/site-packages/pip (python
3.10)
Installed v2 /usr/local/buildpack/tools/v2/python.sh in 17 seconds
skip cleanup, not a docker build: c43d80576233

ERROR: Could not find a version that satisfies the requirement poetry1.1.15
(from versions: none)
ERROR: No matching distribution found for poetry1.1.15

Our renovate.json looks like this:

{
  "constraints": {
    "poetry": "1.1.15"
  },
  "extends": [
    "config:base",
    "docker:enableMajor"
  ],
  "ignorePaths": []
}

What have I got wrong?

**Reply by viceice** [reply]
can you post some more context line from debug log please?

**Comment by anarkiwi** [comment]
I apologize, that's the entire debug output I have (from the renovate PR).

I have a link to the PR's comment if that's helpful:
https://github.com/IQTLabs/poseidon/pull/2714#issuecomment-1262483636

**Reply by viceice** [reply]
you can trigger a rebase from renovate to get new logs. i don't need the PR creation. the issue should also occur on PR rebase

**Comment by anarkiwi** [comment]
Apologies I'm not following... the only place I can find this debug log,
was in the PR body (it's not anywhere in our Actions log), in renovate's
comment "Artifact update problem".

Where should I look for renovate's full debug log?

**Reply by rarkins** [reply]
If you are running with GitHub Actions, logs would be in the actions log. But you may need to export LOG_LEVEL=debug before running if you haven't already
